zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/DecisionDefinitionController.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import io.camunda.service.DecisionDefinitionServices;
+import io.camunda.service.search.query.DecisionDefinitionQuery;
+import io.camunda.zeebe.gateway.protocol.rest.DecisionDefinitionSearchQueryRequest;
+import io.camunda.zeebe.gateway.protocol.rest.DecisionDefinitionSearchQueryResponse;
+import io.camunda.zeebe.gateway.rest.RequestMapper;
+import io.camunda.zeebe.gateway.rest.RestErrorMapper;
+import io.camunda.zeebe.gateway.rest.SearchQueryRequestMapper;
+import io.camunda.zeebe.gateway.rest.SearchQueryResponseMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@ZeebeRestController
+public class DecisionDefinitionController {
+
+  @Autowired private DecisionDefinitionServices decisionDefinitionServices;
+
+  @PostMapping(
+      path = "/decision-definitions/search",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<DecisionDefinitionSearchQueryResponse> searchDecisionDefinitions(
+      @RequestBody(required = false) final DecisionDefinitionSearchQueryRequest query) {
+    return SearchQueryRequestMapper.toDecisionDefinitionQuery(query)
+        .fold(this::search, RestErrorMapper::mapProblemToResponse);
+  }
+
+  private ResponseEntity<DecisionDefinitionSearchQueryResponse> search(
+      final DecisionDefinitionQuery query) {
+    try {
+      final var result =
+          decisionDefinitionServices
+              .withAuthentication(RequestMapper.getAuthentication())
+              .search(query);
+      return ResponseEntity.ok(
+          SearchQueryResponseMapper.toDecisionDefinitionSearchQueryResponse(result));
+    } catch (final Throwable e) {
+      final var problemDetail =
+          RestErrorMapper.createProblemDetail(
+              HttpStatus.BAD_REQUEST,
+              e.getMessage(),
+              "Failed to execute Decision Definition Search Query");
+      return ResponseEntity.of(problemDetail)
```

Comment:
🔧 If I'm not mistaken, the REST error mapper already provides a method to do this.

Reply:
✅